test/test_sink.py

```diff
@@ -176,3 +198,63 @@ def testImageCopySmallMultiband(file_type, tmp_path):
     assert new_metadata.get('levels') == 2 or new_metadata.get('levels') == 3
     assert new_metadata.get('bandCount') == 7
     assert len(new_metadata.get('frames')) == 6
+
+
+@pytest.mark.parametrize('resample_method', RESAMPLE_METHODS)
+def testImageCopyLargeDownsampling(resample_method, tmp_path):
+    output_file = tmp_path / f'{resample_method}.db'
+    sink = large_image_source_zarr.new()
+    source = large_image_source_test.TestTileSource(
+        fractal=True,
+        tileWidth=128,
+        tileHeight=128,
+        sizeX=2048,
+        sizeY=4096,
+        frames='c=2,z=3',
+    )
+    copyFromSource(source, sink)
+    sink.write(output_file, resample=resample_method)
+    written = large_image_source_zarr.open(output_file)
+    written_arrays = dict(written._zarr.arrays())
+
+    assert len(written_arrays) == written.levels
+    assert written_arrays.get('0') is not None
+    assert written_arrays.get('0').shape == (2, 3, 4096, 2048, 3)
+    assert written_arrays.get('1') is not None
+    assert written_arrays.get('1').shape == (2, 3, 2048, 1024, 3)
+    assert written_arrays.get('2') is not None
+    assert written_arrays.get('2').shape == (2, 3, 1024, 512, 3)
+    assert written_arrays.get('3') is not None
+    assert written_arrays.get('3').shape == (2, 3, 512, 256, 3)
+
+    # TODO: Can the content of the downsampled data be compared back to the test source?
+
+
+@pytest.mark.parametrize('resample_method', RESAMPLE_METHODS)
+def testImageCopyLargeDownsamplingMultiband(resample_method, tmp_path):
```

Comment:
These tests are very slow compared to the overall CI time.  There are two different things we are trying to test: that downsampling methods are correct, and that our multi-downsampled levels are correct.  We aren't currently testing all of the second, because the size of these tests aren't big enough to see if we have the overlap effects right.

Probably we should keep these tests, but do them on small samples (just big enough to need downsampling).  Because the fractal is regular, for the numpy tests we can check if we get the expected answer in one or two spots.  The multiband tests could also just be done on a subset of downsampling methods (lanczos and np_nearest_neighbor, maybe) just to test the different code paths.

As a test of the effects of overlap we want to test at just lanczos and np_nearest_neighbor on something more than twice as big as the operating tile size (the 4096 we used), and check (at least visually) that it looks right.  In this case, I probably would only test the single band variant.

The previous test will also test the recursion on the downsampling.

Reply:
Ok, I can decrease the image size of these tests. I will also reduce the set of resample methods sent through the 6-band test.  I will need your help to write the correct assertion for the content of the downsampled levels.
Am I understanding correctly that a new test should use a larger size (but only one band) to test the overlap effects on two resample methods? And this will be the only test that has more than 2 levels?

Reply:
Also, since we are using the small image size for the main downsampling tests, they can be combined with `testImageCopySmall` and `testImageCopySmallMultiband`. I can just move the assertions.

Reply:
That is what I was thinking; I'm trying to balance code coverage and test time.  CI currently takes 20+ minutes on master and is currently failing on this branch because it sometimes exceeds the hour limit.

Reply:
Reorganized the tests in [f7876a3](https://github.com/girder/large_image/pull/1490/commits/f7876a3e4512dc3c2349536a88a5204dca91f13d), including the new assertion for the content of the downsampled levels depending on the resample method.